src/reflect/scala/reflect/runtime/JavaMirrors.scala

```diff
@@ -59,7 +54,7 @@ private[scala] trait JavaMirrors extends internal.SymbolTable with api.JavaUnive
 
   def runtimeMirror(cl: ClassLoader): Mirror = gilSynchronized {
     mirrors get cl match {
-      case Some(WeakReference(m)) => m
+      case Some(mRef) => mRef.get
```

Comment:
I'm not actually sure how to do this correctly without using the extractor (or rolling your own), because there is a race between subsequent calls to `.get`. You'd need something horrible which ends up duplicating the default case.

```scala
case Some(mRef) =>
  val m = mRef.get
  if (m != null) m else createMirror(rootMirror.RootClass, cl) // duplicated logic
```